src/options/parser.rs

```diff
@@ -1,748 +1,115 @@
-                                            return Err(ParseError::NeedsValue { flag, values });
-                                        }
-                                        TakesValue::Optional(_) => {
-                                            result_flags.push((flag, None));
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-
-            // Otherwise, it’s a free string, usually a file name.
-            else {
-                frees.push(arg);
-            }
-        }
-
-        Ok(Matches { frees, flags: MatchedFlags { flags: result_flags, strictness } })
-    }
-
-    fn lookup_short(&self, short: ShortArg) -> Result<&Arg, ParseError> {
-        match self.0.iter().find(|arg| arg.short == Some(short)) {
-            Some(arg)  => Ok(arg),
-            None       => Err(ParseError::UnknownShortArgument { attempt: short })
-        }
-    }
-
-    fn lookup_long(&self, long: &OsStr) -> Result<&Arg, ParseError> {
-        match self.0.iter().find(|arg| arg.long == long) {
-            Some(arg)  => Ok(arg),
-            None       => Err(ParseError::UnknownArgument { attempt: long.to_os_string() })
-        }
-    }
-}
-
-
-/// The **matches** are the result of parsing the user’s command-line strings.
-#[derive(PartialEq, Eq, Debug)]
-pub struct Matches<'args> {
-
-    /// The flags that were parsed from the user’s input.
-    pub flags: MatchedFlags<'args>,
-
-    /// All the strings that weren’t matched as arguments, as well as anything
-    /// after the special “--” string.
-    pub frees: Vec<&'args OsStr>,
-}
-
-#[derive(PartialEq, Eq, Debug)]
-pub struct MatchedFlags<'args> {
-
-    /// The individual flags from the user’s input, in the order they were
-    /// originally given.
-    ///
-    /// Long and short arguments need to be kept in the same vector because
-    /// we usually want the one nearest the end to count, and to know this,
-    /// we need to know where they are in relation to one another.
-    flags: Vec<(Flag, Option<&'args OsStr>)>,
-
-    /// Whether to check for duplicate or redundant arguments.
-    strictness: Strictness,
-}
-
-impl<'a> MatchedFlags<'a> {
-
-    /// Whether the given argument was specified.
-    /// Returns `true` if it was, `false` if it wasn’t, and an error in
-    /// strict mode if it was specified more than once.
-    pub fn has(&self, arg: &'static Arg) -> Result<bool, OptionsError> {
-        self.has_where(|flag| flag.matches(arg))
-            .map(|flag| flag.is_some())
-    }
-
-    /// Returns the first found argument that satisfies the predicate, or
-    /// nothing if none is found, or an error in strict mode if multiple
-    /// argument satisfy the predicate.
+pub use clap::Parser;
+use std::ffi::OsString;
+
+#[derive(Parser)]
```

Comment:
Thinking about it some more, you should probably use `#[derive(Default)]` on `Opts` so that you can call `Opts::default()` in the tests, and you can remove all the boilerplate! `#[derive(Default)]` implements a `default()` method that returns the `struct` with all the values to their default, which means `0` for integer, `false` for boolean, and `None` for `Option`.

So that means:

```suggestion
#[derive(Parser, Default)]
```

And use `Opts::default()` in other files for tests.